Allow for generation of events-api-v2 service integration

mathyourlife-fitbit opened the pull request:
The events v1 api is deprecated.  Adding support to generate events api v2 tokens for services.

> WHEN DOES THE SUPPORT FOR THE PREVIOUS API VERSION END?
  API v1 will be supported until February 6, 2018. If you find yourself in a unique situation that 
  requires an extension, please contact our support team at support@pagerduty.com.
  As of July 6, 2016, no new features will be released for v1.

https://v2.developer.pagerduty.com/v2/docs/api-v2-frequently-asked-questions

Sample curl request:

```bash
curl -X POST 'https://api.pagerduty.com/services/{id}/integrations' \
  -H 'Content-Type: application/json' \
  -H 'Accept: application/vnd.pagerduty+json;version=2' \
  -H 'Authorization: Token token={api token}' \
  --data-binary '{"integration":{"type":"events_api_v2_inbound_integration","name":"events v2 api"}}'
```

heimweh commented:
Hi @mathyourlife-fitbit, thank you so much for this PR! 👍 

This looks good to me (confirmed that it works).
 
Would you mind adding this type to the list of available types in the resource documentation?
it's in `website/docs/r/service_integration.html.markdown`, also it would be worth adding a note that this type requires `alert_creation` to be set to `create_alerts_and_incidents` in the service.

Many thanks,

Alexander

mathyourlife-fitbit commented:
I had already added the option to `website/docs/r/service_integration.html.markdown`.
I added the note that services require the `create_alerts_and_incidents` setting for a v2 api.

heimweh commented:
Hi @mathyourlife-fitbit, sorry I completely missed that. 

This LGTM! 👍 huge thanks!

mathyourlife-fitbit commented:
Not a problem.  Thanks for the quick review, @heimweh .
